vslib/src/VirtualSwitchSaiInterface.cpp

```diff
@@ -602,6 +604,15 @@ std::shared_ptr<SwitchStateBase> VirtualSwitchSaiInterface::init_switch(
             SWSS_LOG_THROW("unable to init switch %s", sai_serialize_status(status).c_str());
         }
 
+        // Initialize switch for VOQ attributes
+
+        status = ss->initialize_voq_switch_objects(attr_count, attr_list);
```

Comment:
ok, this looks better, but i would like you to make 1 more change, so i think it would be actually better to propagate those count/list to parameters initialize_default_objects() function, and call voq initialize inside initialize_devault_objects, this way we will have initialization part in 1 single function, what do you think?

Reply:
No issue in doing what you are proposing. However, here the voq objects are not defaults. They are configured attributes. Not sure if it will be clearer to have non default object initialization in function whose name indicates default object initialization

Reply:
I also see a comment "TODO" to move the calling "initialize_default_objects()" to constructor. If we modify to propagate attr_count/attr_list to this function, we may not be able move this to constructor. Is it still o.k, if I do the modification as proposed?

Reply:
that move to constructor cant be moved to constructor since that is virtual function, it could be moved, but it would need to be called explicitly in each constructor, and not in SwitchStateBase

Reply:
o.k. I'll modify as proposed, then.

Reply:
thanks, this is actually interesting that some of the internal objects depends on the input attributes to create_switch function, and not actual hardware profile, this is wired to me

Reply:
Modified as proposed.